tools/toolchain/scripts/stage6/install_deepmd.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash -e
+
+# TODO: Review and if possible fix shellcheck errors.
+# shellcheck disable=SC1003,SC1035,SC1083,SC1090
+# shellcheck disable=SC2001,SC2002,SC2005,SC2016,SC2091,SC2034,SC2046,SC2086,SC2089,SC2090
+# shellcheck disable=SC2124,SC2129,SC2144,SC2153,SC2154,SC2155,SC2163,SC2164,SC2166
+# shellcheck disable=SC2235,SC2237
+
+[ "${BASH_SOURCE[0]}" ] && SCRIPT_NAME="${BASH_SOURCE[0]}" || SCRIPT_NAME=$0
+SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_NAME")/.." && pwd -P)"
+
+deepmd_ver="2.0.0.b1"
+deepmd_sha256="4860d72ac8ecd485a501a51a8b3e85f1a62ecf477b164ad636b5824a63c83750"
+
+case "$DEEPMD_MODE" in
+  cpu)
+    tfcc_ver="2.3.0-cpu_cudaNone_0"
+    tfcc_sha256="b1b6e5c68a58e516178e6e855a70dd020021d1cf22a94ad291bb8e7541ec58db"
+    ;;
+  cuda)
+    tfcc_ver="2.3.0-gpu_cuda10.1_0"
+    tfcc_sha256="085ba903c76adbbcb57dcdb474e5da5fd36324ca64ce673acacf122d5a2f2b3d"
+    ;;
+esac
+
+cub_link="https://github.com/NVlabs/cub/archive/c3cceac115c072fb63df1836ff46d8c60d9eb304.zip"
+cub_sha256="8894c68d7549681591c34078dcd40cf34459b6c7d33407f07e2145d2adb683ee"
+
+source "${SCRIPT_DIR}"/common_vars.sh
+source "${SCRIPT_DIR}"/tool_kit.sh
+source "${SCRIPT_DIR}"/signal_trap.sh
+source "${INSTALLDIR}"/toolchain.conf
+source "${INSTALLDIR}"/toolchain.env
+
+[ -f "${BUILDDIR}/setup_deepmd" ] && rm "${BUILDDIR}/setup_deepmd"
+
+DEEPMD_LDFLAGS=''
+DEEPMD_LIBS=''
+DEEPMD_CXXFLAGS='-std=gnu++11 '
+
+! [ -d "${BUILDDIR}" ] && mkdir -p "${BUILDDIR}"
+cd "${BUILDDIR}"
+
+case "$with_tfcc" in
+  __DONTUSE__)
+    tensorflow_root="$with_deepmd"
+    check_dir "${tensorflow_root}/include"
+    check_dir "${tensorflow_root}/lib"
+    ;;
+  __INSTALL__)
+    echo "==================== Installing TensorFlow C++ Interface ===================="
+    pkg_install_dir="${INSTALLDIR}/deepmd-kit-${deepmd_ver}"
+    export DP_VARIANT=${DEEPMD_MODE}
+    if [ -f libtensorflow_cc-${tfcc_ver}.tar.bz2 ]; then
+      echo "libtensorflow_cc-${tfcc_ver}.tar.bz2 is found"
+    else
+      download_pkg ${DOWNLOADER_FLAGS} ${deepmd_sha256} \
+        https://anaconda.org/deepmodeling/libtensorflow_cc/2.3.0/download/linux-64/libtensorflow_cc-${tfcc_ver}.tar.bz2
```

Comment:
Installing Tensorflow via Conda is a nice solution given how painful it is to build from source. However, I'm wondering why are you not using the "official" package from [conda forge](https://anaconda.org/conda-forge/libtensorflow_cc)?

Reply:
> Installing Tensorflow via Conda is a nice solution given how painful it is to build from source. However, I'm wondering why are you not using the "official" package from [conda forge](https://anaconda.org/conda-forge/libtensorflow_cc)?

The packages used here are provided by Deep Modeling, the main developer group of DeePMD-kit. I will also try the package on `conda-forge`.

Reply:
> The packages used here are provided by Deep Modeling, the main developer group of DeePMD-kit. I will also try the package on `conda-forge`.

Failed because of the lost of `include/google/protobuf/type.pb.h`. The package on `conda-forge` seems not include some headers needed by DeePMD-kit.

Reply:
That header file is probably part of [libprotobuf](https://github.com/conda-forge/tensorflow-feedstock/blob/f18fb006b85678a5180336e5f99f68a9ba20d5ea/recipe/meta.yaml#L60). So, I guess installing Conda packages by hand is not a very robust solution after all.

I think, then we should keep Tensorflow out of the toolchain for now. It's actually not that hard to install via pip and most HPC centers also provide it.

I've opened #1622 to make Tensorflow available in the CI. Once it's merged you can rebase this PR and continue testing.